### DISCUSSION TITLE
Where should we host helm charts repo of ks-devops

### DISCUSSION THREAD

**Discussion by JohnNiang** [kubesphere/ks-devops#37] [General]
Recently, I was trying to [make devops component separate from ks-installer](https://github.com/kubesphere/ks-installer/pull/1559), but I encountered two problems of how to install ks-devops into ks-installer via helm chart. 

1. Because [ks-devops](https://github.com/LinuxSuRen/ks-devops/tree/93e8b69f02f68d31d84570e564cfc7648a4a0654/charts/ks-devops) charts is hosted into [ks-devops repo](https://github.com/LinuxSuRen/ks-devops/), and has not been released yet. 
2. Another problem we have to consider is offline helm charts support in ks-installer.

So we have to make a plan to host ks-devops charts. There are two potential options for us:

1. [KubeSphere official helm charts](https://github.com/kubesphere/helm-charts)

    #### Pros

    Firstly, we have to separate ks-devops helm charts into a newly repo.

    KubeSphere official has hosted helm charts in [helm-charts repo](https://github.com/kubesphere/helm-charts). We can integrate ks-devops charts into this repo easily by including ks-devops as git submodule, and it can build charts to <https://charts.kubesphere.io/main> or <https://charts.kubesphere.io/test>.

    If ks-devops helm charts is independent, we can use ks-installer in offline mode by including ks-devops helm chart as git submodule.
    
    #### Cons

    We have to separate ks-devops helm charts into a newly repo and integrate it into ks-installer as git submodule. It will increase our maintenance costs.

2. In github page via [chart-releaser-action](https://github.com/helm/chart-releaser-action)

    #### Pros
    
    We can use github action plugin: [chart-releaser-action](https://github.com/helm/chart-releaser-action) to build our charts into gh-page conveniently. Here is a case for reference: <https://github.com/openebs/charts>.

   There is almost no maintenance cost for us.
    
    #### Cons
    
    The accessibility of GitHub Page in mainland China is a big challenge.

But in the end, I prefer choosing first option due to good offline mode support and accessibility of network.

**Comment by JohnNiang** [comment]
For option 2, I have hosted our charts in github page by using [chart-releaser-action](https://github.com/helm/chart-releaser-action). Here is charts repo: <https://johnniang.github.io/ks-devops>.

**Comment by LinuxSuRen** [comment]
Thanks for your thoughts about it. As you mentioned, there're three potential options here. But I didn't see any pros and cons about those options. And you didn't give a preference of yours. It looks like a note instead of a discussion.

**Reply by JohnNiang** [reply]
Hi @LinuxSuRen , I've refined discussion description just now, please take a moment to have a look.
